client/my-sites/domains/domain-management/transfer/transfer-page/index.tsx

```diff
@@ -72,15 +80,36 @@ const TransferPage = ( props: TransferPageProps ) => {
 		isDomainInfoLoading,
 		isDomainLocked,
 		isDomainOnly,
+		isLoadingPurchase,
 		isMapping,
 		isSupportSession,
+		purchase,
 		selectedDomainName,
 		selectedSite,
 	} = props;
 	const { __ } = useI18n();
 	const [ isRequestingTransferCode, setIsRequestingTransferCode ] = useState( false );
 	const [ isLockingOrUnlockingDomain, setIsLockingOrUnlockingDomain ] = useState( false );
+	const [ showUpsellDiscount, setShowUpsellDiscount ] = useState( false );
 	const domain = getSelectedDomain( props );
+	const canUsePreventTransferPromo =
+		domain?.availablePromos?.length && domain?.availablePromos?.length > 0;
+
+	const updateLastTransferOutIntent = async () => {
+		try {
+			await wpcom.req.post(
+				`/me/domains/${ selectedDomainName }/update-last-transfer-out-intent/`,
+				{}
+			);
+			setShowUpsellDiscount( true );
+			recordTracksEvent( 'calypso_show_discount_on_domain_transfer_out', {
```

Comment:
```suggestion
			recordTracksEvent( 'calypso_show_discount_on_domain_transfer_out_intent', {
```
Maybe this is a little more exact?